internal/cmd/testdbman/main.go

```diff
@@ -1,26 +1,403 @@
+runtime.NumCPU() (a choice that comes from pgx's default connection pool size).
+`,
+			createTestDatabases,
+		)
+	}
+
+	// drop
+	{
+		commandBundle.AddCommand(
+			"drop",
+			"Drop test databases",
+			`
+Drops all test databases. Any test database matching the base name
+(river_testdb) or the base name with an underscore followed by any other token
+(river_testdb_example, river_testdb_0, river_testdb_1, etc.) will be dropped.
+`,
+			dropTestDatabases,
+		)
+	}
+
+	// reset
+	{
+		commandBundle.AddCommand(
+			"reset",
+			"Drop and recreate test databases",
+			`
+Reset the test databases, dropping the existing database(s) if they exist, and
+recreating them with the most up to date schema. Equivalent to running "drop"
+followed by "create".
+`,
+			resetTestDatabases,
+		)
+	}
+
+	ctx := context.Background()
+
+	if err := commandBundle.Exec(ctx, os.Args); err != nil {
+		fmt.Fprintf(os.Stderr, "failed: "+err.Error()+"\n")
 		os.Exit(1)
 	}
 }
+
+//
+// Commands
+//
+
+const managementDatabaseURL = "postgres:///postgres"
+
+//
+// Helpers
+//
+
+func createTestDatabases(ctx context.Context, out io.Writer) error {
+	mgmtConn, err := pgx.Connect(ctx, managementDatabaseURL)
+	if err != nil {
+		return fmt.Errorf("error opening management connection: %w", err)
+	}
+	defer mgmtConn.Close(ctx)
+
+	createDBAndMigrate := func(dbName string) error {
+		if _, err := mgmtConn.Exec(ctx, "CREATE DATABASE "+dbName); err != nil {
+			return fmt.Errorf("error crating database %q: %w", dbName, err)
+		}
+		fmt.Fprintf(out, "created: %-20s", dbName)
+
+		// Defer printing a newline, which will be either added to the end of a
+		// successful invocation of this command (after the string "[and
+		// migrated]" has been printed to the current line), or printed before
+		// returning an error so that in either case output looks right.
+		defer fmt.Fprintf(out, "\n")
+
+		dbURL := "postgres:///" + dbName
+
+		dbPool, err := pgxpool.New(ctx, dbURL)
+		if err != nil {
+			return fmt.Errorf("error creating connection pool to %q: %w", dbURL, err)
+		}
+		defer dbPool.Close()
+
+		migrator := rivermigrate.New(riverpgxv5.New(dbPool), nil)
+		if _, err = migrator.Migrate(ctx, rivermigrate.DirectionUp, &rivermigrate.MigrateOpts{}); err != nil {
+			return err
+		}
+		fmt.Fprintf(out, " [and migrated]")
+
+		return nil
+	}
+
+	dbNames := generateTestDBNames(max(4, runtime.NumCPU()))
+
+	for _, dbName := range dbNames {
+		if err := createDBAndMigrate(dbName); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func generateTestDBNames(numDBs int) []string {
+	dbNames := []string{
+		"river_testdb",
+		"river_testdb_example",
+	}
+
+	// This is the same default as pgxpool's maximum number of connections
+	// when not specified -- either 4 or the number of CPUs, whichever is
+	// greater. If changing this number, also change the similar value in
+	// `riverinternaltest` where it's duplicated.
```

Comment:
This comment is probably better placed above line 138 where the `numDBs` is calculated

Reply:
Ah whoops. Good call. Moved.